src/components/Reports/GoalCalculator/SharedComponents/GoalCalculatorGrid/GoalCalculatorGrid.tsx

```diff
@@ -0,0 +1,276 @@
+          .number()
+          .min(0, t('Amount must be positive'))
+          .required(t('Amount is required')),
+      }),
+    ),
+  });
+
+  const handleSubmit = () => {
+    // Handle form submission here
+    // TODO: Implement form submission logic
+    handleContinue();
+  };
+
+  return (
+    <>
+      {promptText && <StyledTypography>{t(promptText)}</StyledTypography>}
+
+      <Formik
+        initialValues={initialValues}
+        validationSchema={validationSchema}
+        onSubmit={handleSubmit}
+        enableReinitialize
+      >
+        {(formikProps: FormikProps<GoalCalculatorGridFormValues>) => (
+          <Form>
+            <GoalCalculatorGridForm
+              formikProps={formikProps}
+              headerName={headerName}
+              ctaText={ctaText}
+            />
+          </Form>
+        )}
+      </Formik>
+    </>
+  );
+};
+
+interface GoalCalculatorGridFormProps {
+  formikProps: FormikProps<GoalCalculatorGridFormValues>;
+  headerName: string;
+  ctaText: string;
+}
+
+const GoalCalculatorGridForm: React.FC<GoalCalculatorGridFormProps> = ({
+  formikProps,
+  headerName,
+  ctaText,
+}) => {
+  const { t } = useTranslation();
+  const { values, setFieldValue } = formikProps;
+
+  const totalAmount = values.gridData.reduce(
+    (sum, item) => sum + item.amount,
+    0,
+  );
+
+  const dataWithTotal = [
+    ...values.gridData,
+    { id: 'total', name: 'Total', amount: totalAmount },
+  ];
+
+  const [directInput, setDirectInput] = useState(false);
+
+  const addExpense = () => {
+    const newId = Math.max(...values.gridData.map((item) => item.id), 0) + 1;
+    const newIncomeItem = {
+      id: newId,
+      name: t('New Income'),
+      amount: 0,
+    };
+    const updatedData = [...values.gridData, newIncomeItem];
+    setFieldValue('gridData', updatedData);
+  };
+
+  const handleDirectInputToggle = () => {
+    setDirectInput(!directInput);
+  };
+
+  const handleDelete = (id: number) => {
+    const updatedData = values.gridData.filter((item) => item.id !== id);
+    setFieldValue('gridData', updatedData);
+  };
+
+  const processRowUpdate = (newRow: GridValidRowModel) => {
+    if (newRow.id === 'total') {
+      return newRow;
+    }
+
+    const updatedData = values.gridData.map((item) =>
+      item.id === newRow.id
+        ? {
+            ...item,
+            name: newRow.name as string,
+            amount: newRow.amount as number,
+          }
+        : item,
+    );
+    setFieldValue('gridData', updatedData);
+    return newRow;
+  };
+
+  const columns: GridColDef[] = [
+    {
+      field: 'name',
+      headerName: t(headerName),
+      flex: 1,
+      minWidth: 200,
+      editable: true,
+    },
+    {
+      field: 'amount',
+      headerName: t('Amount'),
+      flex: 1,
+      minWidth: 150,
+      editable: true,
+      type: 'number',
+      align: 'center',
+      headerAlign: 'center',
+      renderCell: (params) => `$${params.value.toLocaleString()}`,
```

Comment:
We want to use a currency formatter method here with USD and the user's locale.

Reply:
Everything in the goal calculator is in USD. We don't need to take the user's currency preference into account.